src/qudi/hardware/switches/power_switch_keysight.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+
+"""
+Control the Radiant Dyes flip mirror driver through the serial interface.
+
+Copyright (c) 2021, the qudi developers. See the AUTHORS.md file at the top-level directory of this
+distribution and on <https://github.com/Ulm-IQO/qudi-iqo-modules/>
+
+This file is part of qudi.
+
+Qudi is free software: you can redistribute it and/or modify it under the terms of
+the GNU Lesser General Public License as published by the Free Software Foundation,
+either version 3 of the License, or (at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with qudi.
+If not, see <https://www.gnu.org/licenses/>.
+"""
+
+try:
+    import visa
+except ImportError:
+    import pyvisa as visa
+from qudi.core.configoption import ConfigOption
+from qudi.core.statusvariable import StatusVar
+from qudi.util.mutex import RecursiveMutex
+from qudi.interface.switch_interface import SwitchInterface
+
+
+class PowerSwitch(SwitchInterface):
+    """ This class is implements communication with the Keysight E36234A power supply using pyVISA
+
+    Example config for copy-paste:
+
+    power_switch3:
+        module.Class: 'switches.power_switch_keysight.PowerSwitch'
+        options:
+            interface: 'USB0::0x2A8D::0x3402::MY59001224::INSTR'
+            hardware_name: 'Keysight_Power2'  # optional
+            switch_time: 1  # optional
+            remember_states: False  # optional
+            switches:
+                APD: ['OFF', 'ON']  # optional
+                Amplifier: ['OFF', 'ON']  # optional
+    """
+
+    # customize available switches in config. Each switch needs a tuple of at least 2 state names.
+    _switches = ConfigOption(name='switches', missing='error')
+    # optional name of the hardware
+    _hardware_name = ConfigOption(name='hardware_name', default='power_supply', missing='nothing')
+    # if remember_states is True the last state will be restored at reloading of the module
+    _remember_states = ConfigOption(name='remember_states', default=False, missing='nothing')
+    # switch_time to wait after setting the states for the solenoids to react
+    _switch_time = ConfigOption(name='switch_time', default=0.5, missing='nothing')
+    # name of the serial interface where the hardware is connected.
+    # Use e.g. the Keysight IO connections expert to find the device.
+    serial_interface = ConfigOption('interface', 'USB0::0x2A8D::0x3402::MY59001224::INSTR')
+
+    # StatusVariable for remembering the last state of the hardware
+    _states = StatusVar(name='states', default=None)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.lock = RecursiveMutex()

```

Comment:
I think the lock is not needed in this module.

Reply:
These lines are used in all `switch hardwares`. I think we may keep it for now for unification